### PR TITLE
add dbscan

### PR DESCRIPTION
**Pull request by xiaogu-space**
add dbscan

### CODE REVIEW

File: clusters/cluster.go [hunk 1]
```diff
@@ -11,15 +11,18 @@ import (
 
 // A Cluster which data points gravitate around
 type Cluster struct {
-	Center geoos.Point
-	Points Points
+	C      int   //dbscan 用到
+	Points []int //dbscan 用到
```
Comment (coolwxb) [inline clusters/cluster.go:15]: remove comment

File: clusters/dbscan/distance.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package dbscan
+
+import (
+	"math"
+
+	"github.com/spatial-go/geoos"
+	"github.com/spatial-go/geoos/common"
+)
+
+// const (
```
Comment (coolwxb) [inline clusters/dbscan/distance.go:10]: Remove unused code

**Review by coolwxb — CHANGES_REQUESTED** [review]

File: clusters/cluster.go (hunk 1 above)
Comment (xiaogu-space) [reply clusters/cluster.go]: done

File: clusters/dbscan/distance.go (hunk 2 above)
Comment (xiaogu-space) [reply clusters/dbscan/distance.go]: done

**Pull request merged**